sdk/core/core/src/az_span.c

```diff
@@ -77,7 +77,49 @@ AZ_NODISCARD az_result az_span_to_uint64(az_span self, uint64_t* out)
     {
       return AZ_ERROR_PARSER_UNEXPECTED_CHAR;
     }
-    value = value * 10 + d;
+
+    uint64_t const new_value = value * 10 + d;
+    if (new_value < value)
+    {
+      return AZ_ERROR_ARITHMETIC_OVERFLOW;
+    }
+
+    value = new_value;
+  }
+
+  *out = value;
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_span_to_uint32(az_span self, uint32_t* out)
+{
+  int32_t self_length = az_span_length(self);
+  if (self_length <= 0)
```

Comment:
Probably it will never really be, but technically it is possible. I would prefer to limit the code to work only when we know it can work correctly, if that comes at no cost. In this case, if we only check for `== 0`, we know the behavior for '0' and '> 0', but what happens when '< 0'? The behavior is rather undefined.
I don't think it would make a practical difference, so I could write it either way, but prefer <= 0.

Reply:
Sounds good. Maybe `az_span_length` should guarantee the returned length has to be >= 0, by validating on creation of the span.

Reply:
We do validate it, and you should never end up with negative values, unless there's a bug in our code, or if user wrote the code that directly modifies values in the `_internal` struct (which they shouldn't do). But I would still leave the check as `<= 0`, because we don't go that of an extra mile to filter out the subset of input values that does not apply to this function - either `==` or `<=` is one assembly instruction.

I am consider this PR feedback as 'done' for the scope of this PR, and not going to do more commits,and I'm going to resolve this conversation. Please revert it or ping me if you disagree and think that I should not check this in.

Reply:
But if you feel strong about it, I can change it. I just feel we're intentionally introducing undefined behavior where we could easy avoid it. There's an argument to explode as soon as possible, if something's wrong, but I don't think that proceeding with the negative value as normal is going to explode right away.